### PR TITLE
Fix things acting like thrown items when being pulled

### LINKED ISSUE #27604: Click dragging items "tosses them" into bins
## Description
Seemingly click dragging items into disposal bins attempts to "throw" them into the bin and seems to ignore the size of the object as i managed to "toss" a water tank into a bin. 

**Reproduction**
-grab an item, large seems to work best.
-click drag into a disposal bin

**Screenshots**
N/A

**Additional context**
It also would allow people to drag others into disposals with little warning aside from being suddenly dragged.

### LINKED ISSUE #27634: Bladed flatcap effortlessly chops trees when dragged
## Description
By dragging a bladed flatcap and ctrl + right clicking, you can chop a tree in one or two taps of the flatcap.

**Reproduction**
1. Get bladed flatcap
2. Drag/push bladed flatcap onto tree
3. Tree dies

**Screenshots**

https://github.com/space-wizards/space-station-14/assets/44417085/e8b8bb4b-07e5-4150-aa4e-5835b0d78923

**Additional context**
I think it also mispredicts. This could be because it counts as throwing. Other throwable items either don't do enough damage or stick in the tree.

### LINKED ISSUE #27635: You can insert almost anything into disposals.
## Description

You can unanchor the gravity generator and insert it into disposals, as well as probably a bunch of other full machines.

**Reproduction**

Unanchor a machine, drag/push it over disposals and it will insert into disposals and can be flushed.

### LINKED ISSUE #27683: The plunger teleportation
### Bug of the Year 2024 Candidate

1. Pull a plunger and stick it to another person
2. Tell that person to go anywhere
3. Move
4. You have teleported to plunger

https://github.com/space-wizards/space-station-14/assets/124214523/79e832f8-140a-4011-a624-1035c11bae89

### PR DESCRIPTION
**Pull request by notquitehadouken**
## About the PR

Fixes #27683 
Fixes #27635
Fixes #27634
Fixes #27604

i made pulling things not act like you threw them

## Why / Balance

game design

## Media

https://github.com/space-wizards/space-station-14/assets/45953835/c4352125-2a03-4b8b-ab3b-2ff49922096f

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: notquitehadouken (freshlysqueezedpain)
- fix: Pulling people into disposals no longer makes them act like you threw them.

### CODE REVIEW

**Comment by thebadman4662** [comment]
What about throwing them? As in releasing pull for extra range and speed.

**Comment by notquitehadouken** [comment]
> I still don't really understand why pulling was changed to throwing. It might be better to just revert that since it also means ctrl+clicking is faster than just actually pulling since you don't get any slowdown, as well as you can fly through space without a jetpack, and fine placement of items via pulling is now impossible in some circumstances.

old pulling code was an abomination

> What about throwing them? As in releasing pull for extra range and speed.

nope

**Comment by metalgearsloth** [comment]
I'm fine with pullcontroller being re-implemented but the code needs to not be dogshit again.

**Comment by Plykiya** [comment]
> I'm fine with pullcontroller being re-implemented but the code needs to not be dogshit again.

Re-implemented to be more generic?

**Comment by deltanedas** [comment]
pull code using throwing causes so many problems but as usual wyci

**Comment by metalgearsloth** [comment]
> > I'm fine with pullcontroller being re-implemented but the code needs to not be dogshit again.
> 
> Re-implemented to be more generic?

Okay so separately to pulling being split across 4 systems and being a mess, pullcontroller had a list stored on the system for moving pulled stuff and subbed to moveevents on all pulled items which was incredibly slow + did some weird movement stuff. Ideally this would be cleaned up a lot.

**Comment by lzk228** [comment]
also should fix #27683

**Comment by notquitehadouken** [comment]
> also should fix #27683

it does

File: Content.Shared/Throwing/ThrownItemSystem.cs [hunk 1]
```diff
@@ -133,10 +134,15 @@ public void LandComponent(EntityUid uid, ThrownItemComponent thrownItem, Physics
...
         public void ThrowCollideInteraction(ThrownItemComponent component, EntityUid thrown, EntityUid target)
         {
+            if (!component.HitEvents)
+                return;
+
             if (component.Thrower is not null)
                 _adminLogger.Add(LogType.ThrowHit, LogImpact.Low,
                     $"{ToPrettyString(thrown):thrown} thrown by {ToPrettyString(component.Thrower.Value):thrower} hit {ToPrettyString(target):target}.");
 
+            Log.Debug($"{component},{component.HitEvents}:{thrown}:{target}");
```
Comment (EmoGarbage404) [inline Content.Shared/Throwing/ThrownItemSystem.cs:144]: delete

File: Content.Shared/Throwing/ThrownItemComponent.cs [hunk 2]
```diff
@@ -50,5 +50,11 @@ public sealed partial class ThrownItemComponent : Component
         /// </summary>
         [DataField]
         public Vector2? OriginalScale = null;
+
+        /// <summary>
+        ///     If the events on collision will actually be raised
+        /// </summary>
+        [DataField, ViewVariables(VVAccess.ReadWrite), AutoNetworkedField]
+        public bool HitEvents = true;
```
Comment (EmoGarbage404) [inline Content.Shared/Throwing/ThrownItemComponent.cs:58]: this name is legit sinful in it's lack of descriptiveness.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
i don't really think this is a great way of doing this change. I'd much rather someone just make a system or API in physics/engine that handles the act of throwing without the content-side stuff of ThrownItemComponent.

File: Content.Shared/Throwing/ThrownItemComponent.cs (hunk 2 above)
Comment (notquitehadouken) [reply Content.Shared/Throwing/ThrownItemComponent.cs:58]: i like sin

**Comment by notquitehadouken** [comment]
hello doctor freeman

**Comment by AJCM-git** [comment]
this isnt necessary anymore

**Pull request closed without merging**